app/src/main/java/com/github/maleksandrowicz93/cqrsdemo/student/AddStudentCommandHandler.java

```diff
@@ -0,0 +1,51 @@
+package com.github.maleksandrowicz93.cqrsdemo.student;
+
+import com.github.maleksandrowicz93.cqrsdemo.student.dto.AddStudentCommand;
+import com.github.maleksandrowicz93.cqrsdemo.student.dto.StudentData;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import static com.github.maleksandrowicz93.cqrsdemo.student.enums.ResultCode.INVALID_CREDENTIALS;
+import static com.github.maleksandrowicz93.cqrsdemo.student.enums.ResultCode.OK;
+import static com.github.maleksandrowicz93.cqrsdemo.student.enums.ResultCode.STUDENT_ALREADY_EXISTS;
+import static com.github.maleksandrowicz93.cqrsdemo.student.enums.ResultProperty.CONFLICTED_ID;
+
+@Slf4j
+@FieldDefaults(makeFinal = true)
+@RequiredArgsConstructor
+class AddStudentCommandHandler {
+
+    StudentQueryRepository studentQueryRepository;
+    StudentWriteRepository studentWriteRepository;
+    StudentMapper studentMapper;
+    ApiResultFactory<StudentData> resultFactory;
+    SecurityService securityService;
+
+    ApiResult<StudentData> handle(AddStudentCommand command) {
+        var email = command.email();
+        if (StringUtils.isBlank(email)) {
+            log.error("Email should not be blank.");
+            return resultFactory.create(INVALID_CREDENTIALS);
+        }
+        if (StringUtils.isBlank(command.password())) {
+            log.error("Password passed by {} should not be blank.", email);
+            return resultFactory.create(INVALID_CREDENTIALS);
+        }
+        Optional<UUID> id = studentQueryRepository.findStudentIdByEmail(email);
+        if (id.isPresent()) {
```

Comment:
nie jestem purystą programowania funkcyjnego, także tylko notka: można to obsłużyć przez `map` + `orElseGet` i wywołania metod prywatnych